### PR TITLE
Settings not profiles

### PR DESCRIPTION
**Pull request by escattone**
Part of #2449 

This PR is a companion to #3312. It removes the view and edit profile links in the header when signed-in, and also redirects all requests to a user's profile or account to the new Yari `/{locale}/settings` SPA, and adds tests for the redirects.

### CODE REVIEW

File: client/src/ui/molecules/login/index.tsx
Comment (escattone) [inline client/src/ui/molecules/login/index.tsx:69]: Perhaps this should be `Account settings` instead?

Comment (schalkneethling) [reply client/src/ui/molecules/login/index.tsx:69]: You are correct sir.

Comment (peterbe) [reply client/src/ui/molecules/login/index.tsx:69]: Yes please. I still think "Settings" is shorter and better but it doesn't really matter because right now there's no home page for your accounts. 
I had it in my mind that we might soon have an accounts home page. And from there you might go deeper into your account settings.

Comment (peterbe) [inline client/src/ui/molecules/login/index.tsx:69]:
```suggestion
        <a href={`/${locale}/settings`}>Account settings</a>
```

File: client/src/ui/molecules/login/index.tsx
Comment (peterbe) [inline client/src/ui/molecules/login/index.tsx:52]:
```suggestion
```
no need to create a variable if it's only ever used once.

File: testing/integration/headless/test_endpoints.py
Comment (peterbe) [inline testing/integration/headless/test_endpoints.py:138]: So some day this might have to move out of the way and become `/account/settings` or something. Right?

Then, on `/account` that's where we have things like a link to your Settings. A link to your Subscription management, link to FAQ about premium account. etc.

**Review by peterbe — COMMENTED** [review]
Once https://github.com/mdn/yari/pull/3312 lands there will be no code importing `utils/auth-link` anymore. So if we wait for that one to land, merge in `main`, then you can also do a `git rm` on that file.

File: testing/integration/headless/test_endpoints.py
Comment (escattone) [reply testing/integration/headless/test_endpoints.py:138]: @peterbe Probably so, since I think if we use MDN accounts going forward, the account settings page will change or just become part of something bigger. For this PR, I was just concerned with making what we currently have consistent.

Comment (peterbe) [reply testing/integration/headless/test_endpoints.py:138]: The most important thing is that we don't make `/en-US/account/?` a permanent redirect to `/en-US/settings` otherwise we can never get back to using that URI for our accounts home page.

**Review by schalkneethling — APPROVED** [review]

**Comment by peterbe** [comment]
@escattone Would you mind merging in `main` and pushing it one last time. Also, before you push, can you see if anything is using `auth-link` now? If not, git rm the file.

**Comment by peterbe** [comment]
@schalkneethling Would you mind fixing it so that the "Sign out" isn't underlined?
Screen Shot 2021-04-01 at 11 19 38 AM

**Comment by peterbe** [comment]
Also, do you think it's easy to fix that odd black background?

https://user-images.githubusercontent.com/26739/113316606-8793c180-92dc-11eb-8761-7377748f14dc.mov

**Comment by schalkneethling** [comment]
@peterbe sure, as soon as this is merged I can pull and fix. Does that work?

**Comment by peterbe** [comment]
> @peterbe sure, as soon as this is merged I can pull and fix. Does that work?

Sure thing! I think a filing of an issue is more useful than rushing a solution :)
So to put meat to my words, I'll turn my comment into a new issue and assign it to you.

**Comment by peterbe** [comment]
> > @peterbe sure, as soon as this is merged I can pull and fix. Does that work?
> 
> Sure thing! I think a filing of an issue is more useful than rushing a solution :)
> So to put meat to my words, I'll turn my comment into a new issue and assign it to you.

https://github.com/mdn/yari/issues/3415

**Pull request merged**
